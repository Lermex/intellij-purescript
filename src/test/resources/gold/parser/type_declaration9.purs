module Main where

type ALL = forall a m. Semigroup m => (a -> m) -> Array a -> m

type Woop = forall a m. ((Cooli) => String) -> (a -> m) -> Cooli => String