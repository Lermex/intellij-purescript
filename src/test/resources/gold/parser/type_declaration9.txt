Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Main')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('ALL')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('a')
                PsiElement(whitespace)(' ')
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('m')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(ConstrainedType)
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('Semigroup')
                        PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(GenericIdentifier)
                      PsiElement(identifier)('m')
                      PsiElement(whitespace)(' ')
                  PsiElement(=>)('=>')
                  PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PsiElement(()('(')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(GenericIdentifier)
                            PsiElement(identifier)('a')
                            PsiElement(whitespace)(' ')
                        PsiElement(->)('->')
                        PsiElement(whitespace)(' ')
                        PSASTWrapperElement(Type)
                          PSASTWrapperElement(TypeAtom)
                            PSASTWrapperElement(GenericIdentifier)
                              PsiElement(identifier)('m')
                      PsiElement())(')')
                      PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeConstructor)
                          PSASTWrapperElement(Qualified)
                            PsiElement(proper name)('Array')
                            PsiElement(whitespace)(' ')
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(GenericIdentifier)
                          PsiElement(identifier)('a')
                          PsiElement(whitespace)(' ')
                      PsiElement(->)('->')
                      PsiElement(whitespace)(' ')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(GenericIdentifier)
                            PsiElement(identifier)('m')