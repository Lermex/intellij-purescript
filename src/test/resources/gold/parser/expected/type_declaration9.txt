Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('Main')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSTypeDeclarationImpl(TYPE_DECLARATION)
        PsiElement(PSTokenType.type)('type')
        PSTypeListImpl(TYPE_LIST)
          PSDeclarationTypeImpl(DECLARATION_TYPE)
            PsiElement(PSTokenType.PROPER_NAME)('ALL')
        PsiElement(PSTokenType.=)('=')
        PSTypeDeclarationItemImpl(TYPE_DECLARATION_ITEM)
          PSGenericModifierImpl(GENERIC_MODIFIER)
            PsiElement(PSTokenType.forall)('forall')
            PSIdListImpl(ID_LIST)
              PSDeclarationIdImpl(DECLARATION_ID)
                PsiElement(PSTokenType.ID)('a')
              PSIdListImpl(ID_LIST)
                PSDeclarationIdImpl(DECLARATION_ID)
                  PsiElement(PSTokenType.ID)('m')
            PsiElement(PSTokenType..)('.')
          PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
            PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
              PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                PSDeclarationTypeImpl(DECLARATION_TYPE)
                  PsiElement(PSTokenType.PROPER_NAME)('Semigroup')
              PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                PSDeclarationIdImpl(DECLARATION_ID)
                  PsiElement(PSTokenType.ID)('m')
            PsiElement(PSTokenType.=>)('=>')
            PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
              PsiElement(PSTokenType.()('(')
              PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
                PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                  PSDeclarationIdImpl(DECLARATION_ID)
                    PsiElement(PSTokenType.ID)('a')
              PsiElement(PSTokenType.->)('->')
              PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
                PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
                  PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                    PSDeclarationIdImpl(DECLARATION_ID)
                      PsiElement(PSTokenType.ID)('m')
                    PsiElement(PSTokenType.))(')')
                PsiElement(PSTokenType.->)('->')
                PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
                  PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
                    PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                      PSDeclarationTypeImpl(DECLARATION_TYPE)
                        PsiElement(PSTokenType.PROPER_NAME)('Array')
                    PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                      PSDeclarationIdImpl(DECLARATION_ID)
                        PsiElement(PSTokenType.ID)('a')
                  PsiElement(PSTokenType.->)('->')
                  PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
                    PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
                      PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                        PSDeclarationIdImpl(DECLARATION_ID)
                          PsiElement(PSTokenType.ID)('m')
    PSDeclarationsImpl(DECLARATIONS)
      PSDeclarationImpl(DECLARATION)
        PSTypeDeclarationImpl(TYPE_DECLARATION)
          PsiElement(PSTokenType.type)('type')
          PSTypeListImpl(TYPE_LIST)
            PSDeclarationTypeImpl(DECLARATION_TYPE)
              PsiElement(PSTokenType.PROPER_NAME)('Woop')
          PsiElement(PSTokenType.=)('=')
          PSTypeDeclarationItemImpl(TYPE_DECLARATION_ITEM)
            PSGenericModifierImpl(GENERIC_MODIFIER)
              PsiElement(PSTokenType.forall)('forall')
              PSIdListImpl(ID_LIST)
                PSDeclarationIdImpl(DECLARATION_ID)
                  PsiElement(PSTokenType.ID)('a')
                PSIdListImpl(ID_LIST)
                  PSDeclarationIdImpl(DECLARATION_ID)
                    PsiElement(PSTokenType.ID)('m')
              PsiElement(PSTokenType..)('.')
            PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
              PsiElement(PSTokenType.()('(')
              PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
                PsiElement(PSTokenType.()('(')
                PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                  PSDeclarationTypeImpl(DECLARATION_TYPE)
                    PsiElement(PSTokenType.PROPER_NAME)('Cooli')
                  PsiElement(PSTokenType.))(')')
              PsiElement(PSTokenType.=>)('=>')
              PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
                PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
                  PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                    PSDeclarationTypeImpl(DECLARATION_TYPE)
                      PsiElement(PSTokenType.PROPER_NAME)('String')
                    PsiElement(PSTokenType.))(')')
                PsiElement(PSTokenType.->)('->')
                PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
                  PsiElement(PSTokenType.()('(')
                  PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
                    PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                      PSDeclarationIdImpl(DECLARATION_ID)
                        PsiElement(PSTokenType.ID)('a')
                  PsiElement(PSTokenType.->)('->')
                  PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
                    PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
                      PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                        PSDeclarationIdImpl(DECLARATION_ID)
                          PsiElement(PSTokenType.ID)('m')
                        PsiElement(PSTokenType.))(')')
                    PsiElement(PSTokenType.->)('->')
                    PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
                      PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
                        PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                          PSDeclarationTypeImpl(DECLARATION_TYPE)
                            PsiElement(PSTokenType.PROPER_NAME)('Cooli')
                      PsiElement(PSTokenType.=>)('=>')
                      PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
                        PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
                          PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                            PSDeclarationTypeImpl(DECLARATION_TYPE)
                              PsiElement(PSTokenType.PROPER_NAME)('String')